configuration.py

```diff
@@ -0,0 +1,153 @@
+        elif not isinstance(interval, int):
+            self.logger.warn("The parameter {} should be a positive integer, found: {}. Considering the default value as 60 minutes".format(
+                param_name, interval))
+        elif not interval > 0:
+            self.logger.warn("The parameter {} must be a positive integer, found: {}. Considering the default value as 60 minutes".format(
+                param_name, interval))
+
+    def validate(self):
+        self.logger.info('Validating the configuration parameters')
+        # validating mandatory fields
+        if not self.configurations.get('sharepoint.client_id'):
+            print_and_log(self.logger, 'error',
+                          'sharepoint.clien_id cannot be empty')
+            return False
+        if not self.configurations.get('sharepoint.client_secret'):
+            print_and_log(self.logger, 'error',
+                          'sharepoint.client_secret cannot be empty')
+            return False
+        if not self.configurations.get('sharepoint.realm'):
+            print_and_log(self.logger, 'error',
+                          'sharepoint.realm cannot be empty')
+            return False
+        if not self.configurations.get('sharepoint.host_url'):
+            print_and_log(self.logger, 'error',
+                          'sharepoint.host_url cannot be empty')
+            return False
+        if not self.configurations.get('ws.access_token'):
+            print_and_log(self.logger, 'error',
+                          'ws.access_token cannot be empty')
+            return False
+        if not self.configurations.get('ws.source_id'):
+            print_and_log(self.logger, 'error', 'ws.source_id cannot be empty')
+            return False
+        if not self.configurations.get('ws.host_url'):
+            print_and_log(self.logger, 'error', 'ws.host_url cannot be empty')
+            return False
+        if not self.configurations.get('sharepoint.site_collections'):
+            print_and_log(self.logger, 'error',
+                          'sharepoint.site_collections cannot be empty')
+            return False
+
+        # validating enable_document_permissions parameter
+        enable_document_permissions = self.configurations.get(
+            'enable_document_permission')
+        if enable_document_permissions is None:
+            self.logger.warn(
+                "The parameter enable_document_permissions is not present. Considering the default value as yes")
+        elif str(enable_document_permissions).lower() not in ['true', 'false']:
+            self.logger.warn(
+                "The parameter enable_document_permission does not contain a valid value. Expected values are yes/no, found: {}. Considering the default value as yes".format(enable_document_permissions))
+
+        # validating start_time parameter
+        start_time = self.configurations.get('start_time')
+        if not start_time:
```

Comment:
I think you meant `if start_time is None`

Reply:
No longer valid, since we are now using Cerberus for validation